api/server/router/network/network_routes.go

```diff
@@ -275,12 +275,11 @@ func (n *networkRouter) postNetworkConnect(ctx context.Context, w http.ResponseW
 		return err
 	}
 
-	// Always make sure there is no ambiguity with respect to the network ID/name
-	nw, err := n.backend.FindNetwork(vars["id"])
-	if err != nil {
-		return err
-	}
-	return n.backend.ConnectContainerToNetwork(connect.Container, nw.ID(), connect.EndpointConfig)
+	// Unlike other operations, we does not check ambiguity of the name/ID here.
+	// The reason is that, In case of attachable network in swarm scope, the actual local network
+	// may not be available at the time. At the same time, inside daemon `ConnectContainerToNetwork`
```

Comment:
Won't the problem be the same if looking up the network is done daemon-side? `daemon.ConnectToNetwork()` also looks up the network using `FindNetwork()`, which in itself won't _create_ the network on the node? https://github.com/moby/moby/blob/6feae060033544985e548dcf1b9127f8f634fe2b/daemon/container_operations.go#L995-L1016

Reply:
Oh, right so, that's for the "stopped container" case, for the "running container", it eventually comes to `findAndAttachNetwork ()`, which looks to be extending the network; https://github.com/moby/moby/blob/6feae060033544985e548dcf1b9127f8f634fe2b/daemon/container_operations.go#L376-L416

There's a lot of "magic" in those functions; perhaps we should consider moving some of that magic out, and explicitly request a network to be extended, idk